src/abstractprobprog.jl

```diff
@@ -60,3 +60,19 @@ m = decondition(condition(m, obs))
 should hold for generative models `m` and arbitrary `obs`.
 """
 function condition end
+
+""" 
+    AbstractContext
+
+Common base type for Contexts used for dispatching.
+"""
+abstract type AbstractContext end
+
+
+""" 
```

Comment:
@PavanChaggar As @phipsgabler suggested, let's keep the `evaluate!` function in a separate file `evaluate.jl`.

Reply:
shall I keep `AbstractContext` in `abstractprobprog.jl`?

Reply:
That sounds like a better place - I've moved these lines.